packages/language-core/src/generators/template.ts

```diff
@@ -1366,8 +1368,8 @@ export function generate(
 				let camelized = false;
 
 				if (
-					hyphenateAttr(prop.name) === prop.name
-					&& !nativeTags.has(node.tag)
+					canCamelize
+					&& hyphenateAttr(prop.name) === prop.name
```

Comment:
Does it need `|| node.tagType === CompilerDOM.ElementTypes.SLOT`? (I'm not sure yet)